exercises/practice/complex-numbers/tests/Tests.elm

```diff
@@ -0,0 +1,241 @@
+module Tests exposing (tests)
+
+import ComplexNumbers as Complex exposing (Complex)
+import Expect exposing (Expectation, FloatingPointTolerance(..))
+import Test exposing (Test, describe, skip, test)
+
+
+tolerance : FloatingPointTolerance
+tolerance =
+    Absolute 0.00001
+
+
+approximatelyEqual : Complex -> Complex -> Expectation
+approximatelyEqual z1 z2 =
+    Complex.sub z1 z2
+        |> Expect.all
+            [ Complex.real >> Expect.within tolerance 0
+            , Complex.imaginary >> Expect.within tolerance 0
+            ]
+
+
+tests : Test
```

Comment:
Does this work with the test runner? I think it does but can't remember all the details, we should probably add some documentation somewhere!

And does it format the test results in a nice way on the website? I also can't remember where we are at with grouping tests in headings to make this better.

Reply:
Yes, it works fine.
Grouping the tests with the headings is only for concept exercises, this one is a practice exercise, so there is nothing special in the UI. It's also OK to have helper functions like above because the test file is fully visible in the UI.